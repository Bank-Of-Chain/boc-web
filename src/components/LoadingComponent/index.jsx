import React from 'react'

const Loading = props => {
  const { loading, children = '', width = '20px', height = '20px', background, className } = props
  const loadingElement = (
    <svg className={className} style={{ background: background }} width={width} height={height} viewBox="0 0 100 100" preserveAspectRatio="xMidYMid">
      <circle cx="16" cy="50" r="10" fill="#e15b64">
        <animate
          attributeName="r"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="0;0;10;10;10"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="0s"
        ></animate>
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="16;16;16;50;84"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="0s"
        ></animate>
      </circle>
      <circle cx="50" cy="50" r="10" fill="#f47e60">
        <animate
          attributeName="r"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="0;0;10;10;10"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="-0.4629629629629629s"
        ></animate>
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="16;16;16;50;84"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="-0.4629629629629629s"
        ></animate>
      </circle>
      <circle cx="84" cy="50" r="10" fill="#f8b26a">
        <animate
          attributeName="r"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="0;0;10;10;10"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="-0.9259259259259258s"
        ></animate>
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="16;16;16;50;84"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="-0.9259259259259258s"
        ></animate>
      </circle>
      <circle cx="16" cy="50" r="10" fill="#abbd81">
        <animate
          attributeName="r"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="0;0;10;10;10"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="-1.3888888888888888s"
        ></animate>
        <animate
          attributeName="cx"
          repeatCount="indefinite"
          dur="1.8518518518518516s"
          calcMode="spline"
          keyTimes="0;0.25;0.5;0.75;1"
          values="16;16;16;50;84"
          keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
          begin="-1.3888888888888888s"
        ></animate>
      </circle>
    </svg>
  )
  return loading ? loadingElement : children
}

export default Loading
