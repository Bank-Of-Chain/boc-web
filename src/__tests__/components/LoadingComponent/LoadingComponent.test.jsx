import React from 'react'
import { render } from '@testing-library/react'
import LoadingComponent from '@/components/LoadingComponent'

import { describe, expect, it } from 'vitest'

describe('xxxxxxxx', () => {
  it('LoadingComponent Component Render', () => {
    const { asFragment } = render(<LoadingComponent>value</LoadingComponent>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('LoadingComponent Component loading Render', () => {
    const { asFragment } = render(<LoadingComponent loading>value</LoadingComponent>)
    expect(asFragment()).toMatchSnapshot()
  })
})
