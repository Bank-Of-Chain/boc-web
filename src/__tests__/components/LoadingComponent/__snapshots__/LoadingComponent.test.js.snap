// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent Component Render 1`] = `
<DocumentFragment>
  value
</DocumentFragment>
`;

exports[`LoadingComponent Component loading Render 1`] = `
<DocumentFragment>
  <svg
    height="20px"
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
    width="20px"
  >
    <circle
      cx="16"
      cy="50"
      fill="#e15b64"
      r="10"
    >
      <animate
        attributeName="r"
        begin="0s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="0;0;10;10;10"
      />
      <animate
        attributeName="cx"
        begin="0s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="16;16;16;50;84"
      />
    </circle>
    <circle
      cx="50"
      cy="50"
      fill="#f47e60"
      r="10"
    >
      <animate
        attributeName="r"
        begin="-0.4629629629629629s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="0;0;10;10;10"
      />
      <animate
        attributeName="cx"
        begin="-0.4629629629629629s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="16;16;16;50;84"
      />
    </circle>
    <circle
      cx="84"
      cy="50"
      fill="#f8b26a"
      r="10"
    >
      <animate
        attributeName="r"
        begin="-0.9259259259259258s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="0;0;10;10;10"
      />
      <animate
        attributeName="cx"
        begin="-0.9259259259259258s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="16;16;16;50;84"
      />
    </circle>
    <circle
      cx="16"
      cy="50"
      fill="#abbd81"
      r="10"
    >
      <animate
        attributeName="r"
        begin="-1.3888888888888888s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="0;0;10;10;10"
      />
      <animate
        attributeName="cx"
        begin="-1.3888888888888888s"
        calcMode="spline"
        dur="1.8518518518518516s"
        keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1"
        keyTimes="0;0.25;0.5;0.75;1"
        repeatCount="indefinite"
        values="16;16;16;50;84"
      />
    </circle>
  </svg>
</DocumentFragment>
`;
